src/components/CalendarMonth.jsx

```diff
@@ -53,71 +54,93 @@ export function getModifiersForDay(modifiers, day) {
-            <tr key={i}>
-              {week.map((day, j) => {
-                const modifiersForDay = getModifiersForDay(modifiers, day);
-                const className = cx('CalendarMonth__day', {
-                  'CalendarMonth__day--outside': !day || day.month() !== month.month(),
-                }, modifiersForDay.map(mod => `CalendarMonth__day--${mod}`));
-
-                return (
-                  <td className={className} key={j}>
-                    {day &&
-                      <CalendarDay
-                        day={day}
-                        modifiers={modifiersForDay}
-                        onDayMouseEnter={onDayMouseEnter}
-                        onDayMouseLeave={onDayMouseLeave}
-                        onDayMouseDown={onDayMouseDown}
-                        onDayMouseUp={onDayMouseUp}
-                        onDayClick={onDayClick}
-                        onDayTouchStart={onDayTouchStart}
-                        onDayTouchEnd={onDayTouchEnd}
-                        onDayTouchTap={onDayTouchTap}
-                      />
-                    }
-                  </td>
-                );
-              })}
-            </tr>
-          ))}
-        </tbody>
-      </table>
-    </div>
-  );
+export default class CalendarMonth extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      weeks: getCalendarMonthWeeks(props.month, props.enableOutsideDays),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { month, enableOutsideDays } = nextProps;
+    if (!month.isSame(this.props.month)) {
+      this.setState({
+        weeks: getCalendarMonthWeeks(month, enableOutsideDays),
+      });
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState);
+  }
+
+  render() {
+    const {
+      month,
+      monthFormat,
+      orientation,
+      isVisible,
+      modifiers,
+      onDayClick,
+      onDayMouseDown,
+      onDayMouseUp,
+      onDayMouseEnter,
+      onDayMouseLeave,
+      onDayTouchStart,
+      onDayTouchEnd,
+      onDayTouchTap,
+    } = this.props;
+
+    const { weeks } = this.state;
+    const monthTitle = month.format(monthFormat);
+
+    const calendarMonthClasses = cx('CalendarMonth', {
+      'CalendarMonth--horizontal': orientation === HORIZONTAL_ORIENTATION,
+      'CalendarMonth--vertical': orientation === VERTICAL_ORIENTATION,
+    });
+
+    return (
+      <div className={calendarMonthClasses} data-visible={isVisible}>
+        <table>
+          <caption className="CalendarMonth__caption js-CalendarMonth__caption">
+            <strong>{monthTitle}</strong>
+          </caption>
+
+          <tbody className="js-CalendarMonth__grid">
+            {weeks.map((week, i) => (
+              <tr key={i}>
+                {week.map((day, j) => {
+                  const modifiersForDay = getModifiersForDay(modifiers, day);
```

Comment:
If you don't run the calculations for non-visible months, you get weird behavior for transitions. For example, if part of the selected range is in the month before or the month after, it will look unselected during the transition and then flash to the selected teal color.